### PR TITLE
Update middleware sample

### PR DESCRIPTION
**Pull request by Shane32**
- Make the middleware a singleton
- Pulls `ISchema` via DI
- Middleware implements ASP.Net `IMiddleware` interface

### CODE REVIEW

File: src/GraphQL.Harness/Startup.cs [hunk 1]
```diff
@@ -56,6 +56,7 @@ public void ConfigureServices(IServiceCollection services)
             // add infrastructure stuff
             services.AddHttpContextAccessor();
             services.AddLogging(builder => builder.AddConsole());
+            services.AddSingleton<GraphQLMiddleware>();
```
Comment (sungam3r) [inline src/GraphQL.Harness/Startup.cs:59]: How did it work before? I mean ` app.UseMiddleware<GraphQLMiddleware>();` without `services.Add`?

Comment (sungam3r) [reply src/GraphQL.Harness/Startup.cs:59]: All services from ctor and Invaoke are discovered and pulled by ASP.NET subsystem, right?

**Review by sungam3r — APPROVED** [review]

File: src/GraphQL.Harness/Startup.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL.Harness/Startup.cs:59]: Previously, yes that was correct, as it was "ASP.Net Core convention-based middleware".  It was essentially a scoped/transient service, and DI services could be added to either the constructor or Invoke method.

With this change, the middleware is pulled from DI and it is called "ASP.Net Core factory-based middleware".

I think it's just a clearer sample.

Comment (sungam3r) [reply src/GraphQL.Harness/Startup.cs:59]: What about middleware from server repo?

Comment (Shane32) [reply src/GraphQL.Harness/Startup.cs:59]: I'll review when I look at the serialization

**Pull request merged**